src/EventDispatcher/AbstractGuzzleHttpEvent.php

```diff
@@ -0,0 +1,198 @@
+<?php
+namespace M6Web\Bundle\GuzzleHttpBundle\EventDispatcher;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * Class AbstractGuzzleHttpEvent
+ */
+abstract class AbstractGuzzleHttpEvent extends Event
+{
+    const EVENT_NAME = 'm6web.guzzlehttp';
+    const EVENT_ERROR_NAME = 'm6web.guzzlehttp.error';
+
+    /**
+     * Command start time
+     *
+     * @var float
+     */
+    protected $executionStart;
+
+    /**
+     * Command execution time
+     *
+     * @var float
+     */
+    protected $executionTime;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var Response
+     */
+    protected $response;
+
+    /**
+     * @var mixed
+     */
+    protected $reason;
+
+    /**
+     * @var string
+     */
+    protected $clientId;
+
+    /**
+     * Set request
+     *
+     * @param Request $request
+     *
+     * @return static
+     */
+    public function setRequest(RequestInterface $request)
+    {
+        $this->request = $request;
+
+        return $this;
+    }
+
+    /**
+     * Return request
+     *
+     * @return Request
+     */
+    public function getRequest()
+    {
+        return $this->request;
+    }
+
+    /**
+     * Set Response
+     *
+     * @param Response $response
+     *
+     * @return static
```

Comment:
Why `static`?  (Same question for other usage in this PR)
Do you really need the cost of [LSB](https://www.php.net/manual/en/language.oop5.late-static-bindings.php)? Why `self` is not enough?